### ISSUE TITLE
Will controlnet be supported in the future?

### ISSUE THREAD

**Issue by northfoxz** [NVIDIA/Stable-Diffusion-WebUI-TensorRT#66]
Hello,

Will controlnet be supporte din the future? (for 1.5 and SDXL)

Thank you!

**Comment by contentis** [comment]
It's definitely on top of my list of features to implement. But currently, I cannot give an ETA.

As ControlNet is not natively supported in Auto1111, what would be the preferred way of integration?

**Comment by tydia** [comment]
One way I'm thinking of is to first determine whether controlnet extension is installed, if not, abort. If it is present, list all controlnet models and only keep supported models for users to select and convert.

**Comment by contentis** [comment]
AFAIK there are multiple controlnet extensions out there. Ideally, I don't want to support each one of them, is there a most popular / default one?

**Comment by maxious** [comment]
https://github.com/Mikubill/sd-webui-controlnet

**Comment by northfoxz** [comment]
for sd 1.5, I think controlnet-v1-1 is the best
https://huggingface.co/lllyasviel/ControlNet-v1-1

for sdxl, I think diffusers_xl series are the best
https://huggingface.co/lllyasviel/sd_control_collection/tree/main

**Comment by contentis** [comment]
When you say variation, do you mean weights?

**Comment by tydia** [comment]
No I mean all third-party/community control models, which sometimes have different architecture compared to the original controlnets listed above.

FYI, here are some of them: https://huggingface.co/lllyasviel/sd_control_collection

**Comment by anwoflow** [comment]
Exactly, SDXL CN is a little bit of a mess, unlike with 1.5 you have several models for pose, depth etc. 
and on top of it, some are ControlNet LoRA models, which is yet again working differently than T2I models that are also available via CN extension ... just like IPAdapter. 

Overall it would be fantastic if all of them would be supported, but the number of combinations is huge here. At this point, I'm not sure if working directly of the meta data for each 'job' from Automatic wouldn't be a better source of info, rather than checking which extension is installed, with which model/preprocessor variant. Along the lines "do the same job as THIS image", as it lists all the models that were used at least.

**Comment by kanbol** [comment]
> AFAIK there are multiple controlnet extensions out there. Ideally, I don't want to support each one of them, is there a most popular / default one?

i think openpose is the most popular

**Comment by Jonseed** [comment]
> > AFAIK there are multiple controlnet extensions out there. Ideally, I don't want to support each one of them, is there a most popular / default one?
> 
> i think openpose is the most popular

And tile (control_v11f1e_sd15_tile) for upscaling!

**Comment by mykeehu** [comment]
And depth for unknown things

**Comment by MattJeanes** [comment]
I believe they are referring to the actual SD extension, not the different modes of ControlNet

As mentioned previously the most popular one by far is https://github.com/Mikubill/sd-webui-controlnet

**Comment by emperor1412** [comment]
I used softedge_hed and IPAdapter all the time, it would be nice if they can be supported

**Comment by psykokwak-com** [comment]
Oh yes. It could be great.
I use inpaint and openpose.

**Comment by lhucklen** [comment]
Can this still be a thing please?

**Comment by banjaminicc** [comment]
There have been various attempts at ControlNet by 3rd parties, I wonder if they'll be of use to everyone here?

https://www.reddit.com/r/StableDiffusion/s/f2NU603gqj

https://github.com/hnsywangxin/controlnet_stable_tensorrt

https://www.cnblogs.com/wildkid1024/p/17668570.html
https://www.cnblogs.com/wildkid1024/p/17747027.html

https://tianchi.aliyun.com/forum/post/569424

**Comment by wulouzhu** [comment]
Looking forward to it

**Comment by henbucuoshanghai** [comment]
Will controlnet be supported now?

**Comment by lix19937** [comment]
Ref https://github.com/NVIDIA/Stable-Diffusion-WebUI-TensorRT, then modify.
